OneCommunityGlobal/HighestGoodNetworkApp
# Abdel streamlined data input for new members

# Description

#### $\textcolor{red}{\textsf{(PRIORITY HIGH) Jae:}}$: Create a way for easy/nearly-automated data input for creating a new member 
- I’d like to be able to have a way for people to input their own information for initial setup in the software. Right now I set them up manually and it takes me a fair amount of time. It would be better to be able to send them a 1-time link that would have them set up the information for themselves. 
- Link needs to be 1-time so that people can’t just randomly create users
- Link should be generated by me just inputting their email somewhere like the top of the User Management page
- The parts that will need to be set up by me after they set themselves up are their Team and Projects/Tasks, so they shouldn’t have access to choose these. The rest though they should set up themselves. 
- I’d like to get an email each time a person sets themselves up, so I can easily check the details without logging into the software. If errors were made, I could also ask them to correct them by just forwarding the email with the corrections. 

## Related PRS:
- This frontend PR is related to the [#464](https://github.com/OneCommunityGlobal/HGNRest/pull/464) backend PR.

## Main changes explained:
- Updated route.js, userManagment.jsx and usermanagment.css.
- Added setupNewUserPopup.jsx, SetupProfile.jsx, SetupProfileUserEntry.jsx and SetupProfileInvalidToken.jsx.

## How to test:
- check into the current branch.
- do `npm run start:local` to run this PR locally.
- Test for the requirements in the description. 

## Test:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49083865/55f1753d-e3d4-4310-832d-513bed58d269
